src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java

```diff
@@ -0,0 +1,224 @@
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.ActivePowerControl;
+import com.powsybl.iidm.network.extensions.ActivePowerControlAdder;
+import com.powsybl.network.store.iidm.impl.MinMaxReactiveLimitsImpl;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.dto.BatteryModificationInfos;
+import org.gridsuite.modification.server.dto.ReactiveCapabilityCurveModificationInfos;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.MODIFY_BATTERY_ERROR;
+/**
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
+ */
+public class BatteryModification extends AbstractModification {
+
+    private final BatteryModificationInfos modificationInfos;
+
+    private static final String LIMITS = "Limits";
+    private static final String ACTIVE_LIMITS = "Active limits";
+    private static final String SETPOINTS = "Setpoints";
+    public static final String REACTIVE_LIMITS = "Reactive limits";
+
+    public BatteryModification(BatteryModificationInfos modificationInfos) {
+        this.modificationInfos = modificationInfos;
+    }
+
+    private static NetworkModificationException makeBatteryException(String batteryId, String msgSuffix) {
+        return new NetworkModificationException(MODIFY_BATTERY_ERROR, "Battery '" + batteryId + "' : " + msgSuffix);
+    }
+
+    @Override
+    public void check(Network network) throws NetworkModificationException {
+        if (network.getBattery(modificationInfos.getEquipmentId()) != null) {
+            Battery battery = ModificationUtils.getInstance().getBattery(network, modificationInfos.getEquipmentId());
+            if (battery.getReactiveLimits().getKind() == ReactiveLimitsKind.MIN_MAX && (modificationInfos.getMinimumReactivePower() != null || modificationInfos.getMaximumReactivePower() != null)) {
+                checkMaxReactivePowerGreaterThanMinReactivePower(battery);
+            }
+            Collection<ReactiveCapabilityCurve.Point> points = battery.getReactiveLimits().getKind() == ReactiveLimitsKind.CURVE ? battery.getReactiveLimits(ReactiveCapabilityCurve.class).getPoints() : List.of();
+            List<ReactiveCapabilityCurve.Point> batteryPoints = new ArrayList<>(points);
+            List<ReactiveCapabilityCurveModificationInfos> modificationPoints = modificationInfos.getReactiveCapabilityCurvePoints();
+            if (!CollectionUtils.isEmpty(points) && modificationPoints != null) {
+                checkMaxQGreaterThanMinQ(batteryPoints, modificationPoints);
+            }
+        }
+    }
+
+    private void checkMaxQGreaterThanMinQ(List<ReactiveCapabilityCurve.Point> batteryPoints, List<ReactiveCapabilityCurveModificationInfos> modificationPoints) {
```

Comment:
I tried but it's a bit complicated to mutualise this method since we have an exception in a foreach loop ...
except that you have suggestions for that

Reply:
You could pass additional parameters to the check method checkMaxQGreaterThanMinQ :

modificationInfos.getEquipmentId() : String
exceptionType : NetworkModificationException.Type (MODIFY_GENERATOR_ERROR or MODIFY_BATTERY_ERROR)
errorMessage : String

and use it in BatteryModification and GeneratorModification ...

Similar thing in check method checkMaxReactivePowerGreaterThanMinReactivePower with additional parameters :

MinMaxReactiveLimits object instead of Generator or Battery
exceptionType : NetworkModificationException.Type (MODIFY_GENERATOR_ERROR or MODIFY_BATTERY_ERROR)
errorMessage : String

Reply:
done